torchvision/transforms/v2/functional/_type_conversion.py

```diff
@@ -25,3 +29,176 @@ def to_image(inpt: Union[torch.Tensor, PIL.Image.Image, np.ndarray]) -> tv_tenso
+    """
+    import nvcv  # type: ignore[import-not-found]
+
+    num_channels = img_tensor.shape[2]
+    dtype = img_tensor.dtype
+
+    # Handle single channel images
+    if num_channels == 1:
+        img_tensor = img_tensor[:, :, 0]
+        if dtype == torch.uint8:
+            return nvcv.Format.U8, img_tensor
+        elif dtype == torch.int16:
+            return nvcv.Format.S16, img_tensor
+        elif dtype == torch.int32:
+            return nvcv.Format.S32, img_tensor
+        elif dtype == torch.float32:
+            return nvcv.Format.F32, img_tensor
+        else:
+            raise TypeError(f"Unsupported dtype {dtype} for single channel image")
+
+    # Handle 3 channel images (defaults to RGB)
+    elif num_channels == 3:
+        if dtype == torch.uint8:
+            return nvcv.Format.RGB8, img_tensor
+        elif dtype == torch.float32:
+            return nvcv.Format.RGBf32, img_tensor
+        else:
+            raise TypeError(f"Unsupported dtype {dtype} for 3-channel image")
+
+    raise ValueError(f"Only 1 and 3 channel images are supported. Got {num_channels} channels.")
+
+
+@torch.jit.unused
+def to_nvcv_tensor(pic) -> "nvcv.Tensor":
+    """Convert a torch.Tensor to nvcv.Tensor. This function does not support torchscript.
+
+    See :class:`~torchvision.transforms.v2.ToNVCVTensor` for more details.
+
+    Args:
+        pic (torch.Tensor): Image to be converted to nvcv.Tensor.
+            Tensor can be in CHW format (unbatched) or NCHW format (batched).
+            Only 1-channel and 3-channel images are supported.
+
+    Returns:
+        nvcv.Tensor: Image converted to nvcv.Tensor with NHWC layout.
+    """
+    import cvcuda  # type: ignore[import-not-found]
+    import nvcv  # type: ignore[import-not-found]
+
+    if not torch.jit.is_scripting() and not torch.jit.is_tracing():
+        _log_api_usage_once(to_nvcv_tensor)
+
+    # Validate input type
+    if not isinstance(pic, torch.Tensor):
+        raise TypeError(f"pic should be `torch.Tensor`. Got {type(pic)}.")
+
+    # Handle different tensor formats and track if input was batched (NCHW) or unbatched (CHW/HW)
+    if pic.ndim == 4:
+        # Batched tensor in NCHW format, permute to NHWC
+        img_tensor = pic.permute(0, 2, 3, 1)
+        input_was_batched = True
+    elif pic.ndim == 3:
+        # Unbatched tensor in CHW format, permute to HWC
+        img_tensor = pic.permute(1, 2, 0)
+        input_was_batched = False
+    else:
+        # 2D or other formats (unbatched single-channel)
+        img_tensor = pic
+        input_was_batched = False
+
+    # Ensure image has channel dimension for unbatched case
+    if img_tensor.ndim == 2:
+        img_tensor = img_tensor.unsqueeze(2)  # H W -> H W C
+
+    # Validate dimensions
+    if img_tensor.ndim not in (3, 4):
+        raise ValueError(f"pic should be 2/3/4 dimensional. Got {img_tensor.ndim} dimensions.")
+
+    # For batched inputs, use the first image to infer format
```

Comment:
Below: just add a fake batch dim to the 3D input so that it becomes 4D. Then you can assume it's always 4D and avoid all the if/else checks.

Although I am almost tempted to not support 3D inputs. CV-CUDA will mostly be faster for large batch sizes, so supporting 3D unbatched data adds little value. We could add it later if needed and keep things simpler for now. Thoughts?